### DISCUSSION TITLE
Why doesn't Github syntax highlight this shell script?

### DISCUSSION THREAD

**Discussion by sparr** [community/community#132425] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

https://github.com/sparr/dotfiles/blob/main/.aliases

I'm not sure what all heuristics are used to identify bash/shell scripts without .sh extensions, but this file seems pretty obviously a shell script so I'm curious why it doesn't get detected. it has `#` comment lines, `shellcheck` commanets, a `for do done` structure, `alias` and `command` commands, command line operators like `&>` and `&&`, etc.

**Comment by ryantusi** [comment]
Hey Sparr!

GitHub's syntax highlighting relies on the Linguist library to determine the language of a file. Linguist uses various heuristics to detect the language like you mentioned, the main are:

- **File Extension:** The primary way Linguist detects file type is by the file extension. Files with extensions like `.sh`, `.bash`, or `.zsh` are recognized as shell scripts.

-  **Shebang Line:** Files that start with a shebang line (e.g., `#!/bin/bash or #!/usr/bin/env bash`) are recognized as shell scripts.

- **Content Heuristics:** Linguist also analyzes the content of files to make a determination. This includes recognizing specific syntax structures, keywords, and comments typical to a language.

If your shell script doesn't have a recognized extension or a shebang line, GitHub might not automatically highlight it as a shell script. I'll give you some ideas you can follow to eradicate this issue:

-  **Add a Shebang Line:** If your script doesn't already have one, add a shebang line at the top of the file. For example: `#!/bin/bash`

- **Use a Recognized Extension:** Rename your file to include a `.sh` extension. For example, `myscript` can be renamed to `myscript.sh`.

- **Explicitly Set the Language in a Markdown File:** If your script is being embedded in a Markdown file, you can explicitly set the language for syntax highlighting:
```
  ```sh
  # Your shell script here
```

- **Check GitHub Linguist Overrides:** You can create a `.gitattributes` file in your repository to manually specify the language of specific files. For example: `your_script_name linguist-language=Shell`


I guess any one of these might rectify your issue, you can improve the "chances" (not guaranteed though) of your shell script being properly highlighted on GitHub. If the issue persists, Github needs to answer this LOL.
Here is a sample `.gitattributes` file to force GitHub to recognize a file as a shell script:
`myscript linguist-language=Shell`

**_Happy Coding and also if you can, please approve this response & give a follow, it helps me get recognition, thanks._**

**Comment by sparr** [comment]
Thanks. I'm going to check out https://github.com/github-linguist/linguist and run it on this file and see if I can debug how it's failing the heuristics so I can make a PR over there to make it more robust.
